### PR TITLE
README.md update

### PR DESCRIPTION
**Pull request by mpergler**
Update README.md to be more visual, and give succint but comprehensive functionality overview.
Per discussion at http://osgeo-org.1560.x6.nabble.com/QGIS-Developer-A-nice-task-for-a-community-member-revamping-the-QGIS-github-landing-page-tt5457164.html,  thread started by @nyalldawson 

Update done jointly by myself and @benhur07b, with text on Server by @jgrocha. Incorporates comments by a few others.

File ./README.md updated, as well as source images added into ./images/README-md/

### CODE REVIEW

**Comment by nyalldawson** [comment]
Looking beautiful :star_struck: thanks everyone for your efforts here!

File: README.md [hunk 1]
```diff
@@ -1,98 +1,187 @@
...
+## Features 
+
+### 1. Flexible and powerful spatial data management
+
+- Supports raster, vector, mesh, and point cloud data in a range of industry-standard formats
+    - *Raster formats include:* GeoPackage, GeoTIFF, GRASS, ArcInfo binary and ASCII grids, ERDAS Imagine SDTS, WMS, WCS, PostgreSQL/PostGIS, and [other GDAL supported formats](https://gdal.org/drivers/raster/index.html).
+    - *Vector formats include:* GeoPackage, ESRI Shapefiles, GRASS, SpatiaLite, PostgreSQL/PostGIS, MSSQL, Oracle, WFS, Vector Tiles and [other OGR supported formats](http://www.gdal.org/ogr_formats.html). 
+    - *Mesh formats include*: NetCDF, GRIB, 2DM, and [other MDAL supported formats](https://github.com/lutraconsulting/MDAL#supported-formats).
+    - *Point-cloud format*: LAS/LAZ and EPT datasets.
+- Access and display local files, spatial databases (PostGIS, SpatiaLite, MSSQL, SAP HANA), web services (WMS, WCS, WFS, ArcGIS REST services), tile services, etc. <!-- removed DB2 due to https://github.com/qgis/QGIS/pull/41178 -->
```
Comment (nyalldawson) [inline README.md:21]:
```suggestion
- Access and display local files, spatial databases (PostGIS, SpatiaLite, SQL Server, Oracle, SAP HANA), web services (WMS, WCS, WFS, ArcGIS REST services), tile services, etc. <!-- removed DB2 due to https://github.com/qgis/QGIS/pull/41178 -->
```

File: README.md [hunk 2]
```diff
@@ -1,98 +1,187 @@
...
+-->
+
+### 5. QGIS server
+
+Headless map server -- running on Linux, macOS, Windows, or in a docker container -- that shares the same code base as QGIS.
+
+- Industry-standard protocols (WMS, WFS, WFS3/OGC API for Features and WCS) allow plug-n-play with any software stack
+- Works with any web server (Apache, nginx, etc) or standalone
+- All QGIS beautiful cartography is supported with best-in-class support for printing
+- Fully customizable with Python scripting support
```
Comment (nyalldawson) [inline README.md:98]:
```suggestion
- All beautiful QGIS cartography is supported with best-in-class support for printing
- Fully customizable with Python scripting support
```

File: README.md [hunk 3]
```diff
@@ -1,98 +1,187 @@
...
+
+- [Training Manual](https://docs.qgis.org/latest/en/docs/training_manual/index.html)
+- [QGIS User Guide](https://docs.qgis.org/latest/en/docs/user_manual/index.html)
+- [QGIS Server Guide](https://docs.qgis.org/latest/en/docs/server_manual/index.html)
+- [Documentation Guidelines](https://docs.qgis.org/latest/en/docs/documentation_guidelines/index.html)
+- [PyQGIS Cookbook](https://docs.qgis.org/latest/en/docs/pyqgis_developer_cookbook/index.html)
+- [Developers Guide](https://docs.qgis.org/latest/en/docs/developers_guide/index.html)
+- [Visual Changelog](https://qgis.org/en/site/forusers/visualchangelogs.html)
+
+### Help and Support Channels
```
Comment (nyalldawson) [inline README.md:152]:
```suggestion
### Help and support channels
```

**Comment by nirvn** [comment]
This is great, huge service to the project. Thanks

File: README.md [hunk 4]
```diff
@@ -1,98 +1,187 @@
...
+For large-scale changes, you can open a [QEP (QGIS Enhancement Proposal)](https://github.com/qgis/QGIS-Enhancement-Proposals). QEPs are used in the process of creating and discussing new enhancements or policy for QGIS. 
 
-### Building from source
+### Translations
 
-The [building guide](INSTALL.md) can be used to get started with building QGIS from source.
+Please help translate QGIS to your language. At this moment about forty languages are already available in the Desktop user interface and about eighty languages are available in transifex ready to be translated.
+
+The [translation](https://qgis.org/en/site/getinvolved/translate.html) process is managed by the [Translation Team](https://qgis.org/en/site/getinvolved/governance/governance.html#gui-translation) and all the activities are done under the [Transifex](https://www.transifex.com/) platform.
+
```
Comment (DelazJ) [inline README.md:184]: Is it worth to mention that every language that is translated at 35% is automatically pulled in the release? People often wonder why their language is not pulled or not pulled anymore.

File: README.md [hunk 5]
```diff
@@ -1,98 +1,187 @@
...
+- [Documentation Guidelines](https://docs.qgis.org/latest/en/docs/documentation_guidelines/index.html)
+- [PyQGIS Cookbook](https://docs.qgis.org/latest/en/docs/pyqgis_developer_cookbook/index.html)
+- [Developers Guide](https://docs.qgis.org/latest/en/docs/developers_guide/index.html)
+- [Visual Changelog](https://qgis.org/en/site/forusers/visualchangelogs.html)
+
+### Help and support channels
+
+There are several channels where you can find help and support for QGIS:
+
+- Using the QGIS community site at [https://qgis.org](https://qgis.org)
```
Comment (DelazJ) [inline README.md:156]:
```suggestion
- Using the QGIS community site at https://qgis.org
```

File: README.md [hunk 6]
```diff
@@ -1,98 +1,187 @@
...
+### Translations
 
-The [building guide](INSTALL.md) can be used to get started with building QGIS from source.
+Please help translate QGIS to your language. At this moment about forty languages are already available in the Desktop user interface and about eighty languages are available in transifex ready to be translated.
+
+The [translation](https://qgis.org/en/site/getinvolved/translate.html) process is managed by the [Translation Team](https://qgis.org/en/site/getinvolved/governance/governance.html#gui-translation) and all the activities are done under the [Transifex](https://www.transifex.com/) platform.
+
+### Other ways to contribute
+
+If you are not a developer, there are many other possibilities which do not require programming skills to help QGIS to evolve. Check our [project homepage for more information](http://qgis.org/en/site/getinvolved/index.html).
```
Comment (DelazJ) [inline README.md:187]:
```suggestion
If you are not a developer, there are many other possibilities which do not require programming skills to help QGIS to evolve. Check our [project homepage for more information](https://qgis.org/en/site/getinvolved/index.html).
```
It would be really nice that one day we write about how a person can contribute to such a project, as discussed in a thread few months ago.

File: README.md [hunk 7]
```diff
@@ -1,98 +1,187 @@
...
-means that QGIS feels snappy to use and has a pleasing, easy to use graphical
-user interface.
-
-QGIS aims to be an easy to use GIS, providing common functions and
-features. The initial goal was to provide a GIS data viewer. QGIS has
-reached that point in its evolution and is being used by many for their
-daily GIS data viewing and editing needs. QGIS supports a number of raster
-, vector and mesh data formats, with new support easily added using the plugin
-architecture.
+# About QGIS
```
Comment (DelazJ) [inline README.md:8]: How about using https://freelance-tech-writer.github.io/table-of-contents-generator/index.html to generate a table of contents at the beginning, as done in the install.md file?

File: README.md [hunk 8]
```diff
@@ -1,98 +1,187 @@
...
+![Example: Style manager](https://docs.qgis.org/latest/en/_images/stylemanager.png "Style Manager")
+
+*Example: Plugins*
+
+![Example: Plugins](images/README-md/plugins_1.png "Plugins")
+
+<!-- Kill this one for now, since it's Python2 not 3
+Example: Python console
+
+![Example: Python console](https://docs.qgis.org/latest/en/_images/python_console_editor.png "Python console")
```
Comment (DelazJ) [inline README.md:88]: we'll be happy to get this fixed in the docs :smile:

File: README.md [hunk 9]
```diff
@@ -1,98 +1,187 @@
...
+![Example: graphical model](https://docs.qgis.org/latest/en/_images/models_model.png "Graphical model")
+
+### 4. Powerful customization and extensibility
+
+- Fully customizable user experience, including user interface and application settings that cater to power-users and beginners alike
+- Rich [expression engine](https://docs.qgis.org/testing/en/docs/user_manual/working_with_vector/expression.html) for maximum flexibility in visualization and processing
+- Broad and varied [plugin ecosystem](https://plugins.qgis.org/) that includes data connectors, digitizing aids, advanced analysis and charting tools, in-the-field data capture, etc.
+- Style manager for creating, storing, and managing styles
+- [QGIS style hub](https://plugins.qgis.org/styles/) for easy sharing of styles
+- Python and C++ API for standalone (headless) applications as well as in-application comprehensive scripting (PyQGIS)
```
Comment (DelazJ) [inline README.md:75]: Link the pyqgis and c++ api website here, or are they mentioned somewhere else?

**Review by DelazJ — COMMENTED** [review]
Agreed. Very nice. Thanks.

File: README.md (hunk 5 above)
Comment (Houska1) [reply README.md:156]: Not sure what this change adds? Parallelism with other links isn't bad. But happy to change if important.

File: README.md (hunk 6 above)
Comment (Houska1) [reply README.md:187]: Agreed...ideas include expanding on documentation (which is great, but a bit dry, "let's make sure we have a sentence about every added feature"; could use lots more explanation of how and why), adding to the new style hub, encouraging people to provide support in the channels mentioned elsewhere....but let's not have this discussion in this thread.

File: README.md (hunk 7 above)
Comment (Houska1) [reply README.md:8]: Good idea. Let's get this pull committed and do this in next round. I'm hoping people come up with even better visual examples to rotate in too.

File: README.md (hunk 8 above)
Comment (Houska1) [reply README.md:88]: :) -- though not sure we need to include it here anyway. Think best to limit to 2 examples per section, and python code is python code. We threw this one off the island because of the 2 vs 3, but think we ended up with a better set of 2 complementary ex anyway.

File: README.md (hunk 9 above)
Comment (Houska1) [reply README.md:75]: Good idea will add.

File: README.md [hunk 10]
```diff
@@ -1,98 +1,187 @@
...
+
+QGIS is developed using the [Qt toolkit](https://qt.io) and C++, since 2002, and has a pleasing, easy to use graphical
+user interface with multilingual support. It is maintained by an active developer team and supported by vibrant 
+community of GIS professionals and enthusiasts as well as geospatial data publishers and end-users. 
+
+### Versions and release cycle
+
+QGIS development and releases follow a [time based schedule/roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html). There are three main branches of QGIS that users can install. These are the **Long Term Release (LTR)** branch, **Latest Release (LR)** branch, and the **Development (Nightly)** branch. 
+
+Every month, there is a **Point Release** that provides bug-fixes to the LTR and bug-fixes as well as new features to the LR.
```
[the remark this answers is not in the document]
Comment (Houska1) [reply README.md:118]: Could have philosophical/purist discussion about it, but I think we've been adding enough enhancements into point releases that it's valid as is.

File: README.md [hunk 11]
```diff
@@ -1,98 +1,187 @@
...
+Headless map server -- running on Linux, macOS, Windows, or in a docker container -- that shares the same code base as QGIS.
+
+- Industry-standard protocols (WMS, WFS, WFS3/OGC API for Features and WCS) allow plug-n-play with any software stack
+- Works with any web server (Apache, nginx, etc) or standalone
+- All beautiful QGIS cartography is supported with best-in-class support for printing
+- Fully customizable with Python scripting support
+
+*Example: QGIS server WMS response*
+
+![Example: QGIS Server response to a WMS request](https://docs.qgis.org/latest/en/_images/server_selection_parameter.png "QGIS Server response to a WMS request")
```
[the remark this answers is not in the document]
Comment (Houska1) [reply README.md:102]: Good point. I'm torn between making everything self-contained in this repo for greater stability, but then it sits and slowly ages (like the Python 2 vs 3 script screenshot in the docs), vs reuse. In the balance, for now we decided to feel free to point to images on our own website, plus the QGIS flickr group which seems stable enough we scrape it at https://www.qgis.org/en/site/about/screenshots.html already. But we pulled other images into ./images/README-md/

Bottom line is if our own links rot, we'll notice pretty quickly and it's easy to fix. And I'd love to get this into master this way to provoke people to come up with even better examples we can then reuse in docs. If we continue to be worried about this, any image can be copied to ./images/README-md/ whenever desired.

File: README.md (hunk 9 above)
Comment (Houska1) [reply README.md]: Done

**Comment by Houska1** [comment]
Thanks for all the various edits/suggestions/typo fixes. All the small ones addressed. Let's merge this pull request to make this go live. I'm hoping having it in place will provoke further suggestions, as well as ideally even better visual examples being provided over time. In a next iteration can incorporate @DelazJ's suggestions of including a TOC and putting even our own qgis.org-sourced images in ./images/README-md/ to avoid future link rot.

If anyone feels like completing the merge, don't think need to wait (yet again!) for the 20.04 and 20.10 compile and build checks to complete in this instance :)

File: README.md [hunk 12]
```diff
@@ -1,98 +1,187 @@
...
+You can help us **by submitting bug reports or fixing bugs** in the [QGIS bug tracker](https://github.com/qgis/QGIS/issues/).
 
-QGIS is on GitHub at https://github.com/qgis/QGIS. If you wish to contribute
-patches you can [fork the project](https://help.github.com/forking/), make your changes, commit to your
-repository, and then [create a pull request](https://help.github.com/articles/creating-a-pull-request-from-a-fork/). The development team can then review your contribution and commit it upstream as appropriate.
+### New features and enhancements
+
+If you wish to contribute patches you can [fork the project](https://help.github.com/forking/), make your changes, commit to your repository, and then [create a pull request](https://help.github.com/articles/creating-a-pull-request-from-a-fork/). The development team can then review your contribution and commit it upstream as appropriate.
 
 If you commit a new feature, add [FEATURE] to your commit message AND give a clear description of the new feature. A webhook will automatically create an issue on the QGIS-Documentation repo to tell people to write documentation about it.
```
[the remark this answers is not in the document]
Comment (Houska1) [reply README.md:182]: Sorry, not following exactly what change is needed here. I'll invite you as contributor to my branch/fork and you can adjust this (and other things) directly there, if you wish, or post-merge in master of course.

File: README.md (hunk 4 above)
Comment (Houska1) [reply README.md:191]: Interesting. Feels to me like it's going a bit too much into the weeds to put this into the overall README.md, but seems like a good point for the "how to contribute effectively if you're not a developer" landing page to be created (somewhere?) as per comment elsewhere in this discussion.

File: README.md (hunk 7 above)
Comment (Houska1) [reply README.md:8]: @benhur07b tells me he explored this, but it pushes the features section further down the page. Think our goal is to get that and the visuals in front of eyeballs as close to the top as possible.

File: README.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (DelazJ) [reply README.md:102]: OK. Let's keep as is and hope we catch things when/if they break.

File: README.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (DelazJ) [reply README.md:118]: > I think we've been adding enough enhancements into point releases that it's valid as is.

I'm not sure of this and even then, it's not deliberate (and are they not ported to LTR?). IMO it looks confusing to expose this kind of info to users. Afaiu, new feature = next release. Any released version only gets fixes, either for 3 months (LR) or for 15 (LTR).

File: README.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (DelazJ) [reply README.md:182]: I did not make a suggestion because I failed to find the right balance and wondered if it should be removed. Here's a suggestion anyway.
```suggestion
If you commit a new feature, add `[FEATURE]` to your commit message AND give a clear description of the new feature. The label `Needs documentation` will be added by maintainers and will automatically create an issue on the QGIS-Documentation repo to tell people to write documentation about it.
```
But I don't like the "tell people to write documentation" part since it may convey the message that the developer is not welcome to doc his work himself (there's already people to do that).

File: README.md (hunk 5 above)
Comment (DelazJ) [reply README.md:156]: Maybe, parallelism could be 
```suggestion
- Using the QGIS community [site](https://qgis.org)
```
Looks weird to me to encapsulate a hyperlink within a hyperlink. Users would anyway read the url in this case so...
bref... no strong opinion.

File: README.md (hunk 7 above)
Comment (DelazJ) [reply README.md:8]: I didn't try myself but maybe it depends on how deep we want to expand the toc

File: README.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (Houska1) [reply README.md]: OK. I simplified to "provides bug-fixes to the LTR and LR".

File: README.md (hunk 5 above)
Comment (Houska1) [reply README.md]: OK. I like that. fixed.

File: README.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (Houska1) [reply README.md]: Ah, now I understand. I just put in the following, based on your suggestion but inviting documentation by the contributor:

> The label `Needs documentation` will be added by maintainers and will automatically create an issue on the QGIS-Documentation repo, where you or others should write documentation about it.

File: README.md (hunk 5 above)
Comment (Houska1) [reply README.md]: Fixed with [QGIS community site](...)

File: README.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (DelazJ) [reply README.md]: Nicer

**Review by DelazJ — APPROVED** [review]

**Comment by roya0045** [comment]
@Houska1 @benhur07b Good to see this lands, really freshen up the landing page!

**Comment by benhur07b** [comment]
Thank you to everyone who helped, reviewed, and gave suggestions. Glad to have worked on this and looking forward to contributing more to the project.
